lib/src/native/realm_core.dart

```diff
@@ -1016,13 +1017,34 @@ class _RealmCore {
     });
   }
 
+  static void _logCallback(Pointer<Void> userdata, int levelAsInt, Pointer<Int8> message) {
+    try {
+      final logger = userdata.toObject<Logger>(isPersistent: true)!;
+      final level = LogLevel.values[levelAsInt].loggerLevel;
+
+      // Don't do expensive utf8 to utf16 conversion unless we have to..
+      if (logger.isLoggable(level)) {
```

Comment:
This one will work but if we do if..else we can remove the comments.

if (logger.isLoggable()) {
logger.log(toRealmDartString(freeRealmMemory: true))
}
else {
realm_free();
}

Reply:
I prefer as is. The comment is really about the cost of moving from core (UTF-8) to dart (UTF-16)